lldb/test/API/functionalities/gdb_remote_client/TestReadMemory.py

```diff
@@ -0,0 +1,36 @@
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+from lldbsuite.test.gdbclientutils import *
+from lldbsuite.test.lldbgdbclient import GDBRemoteTestBase
+
+
+class TestReadMemory(GDBRemoteTestBase):
+    def test_x_with_prefix(self):
+        class MyResponder(MockGDBServerResponder):
+            def qSupported(self, client_features):
+                return super().qSupported(client_features) + ";binary-upload+"
+
+            def x(self, addr, length):
+                return "bfoobar" if addr == 0x1000 else "E01"
```

Comment:
If you wanted to be even more clear, you could stub `m` as well and error. Test would fail anyway if we did call `m` though, it won't get the right value back.

Reply:
Oh right yeah, I've had it stall for ages while it realises what happened. Best avoided then.